### ISSUE TITLE
Inconsistent implementation on adjust_hue with PIL

### ISSUE THREAD

**Issue by shijianjian** [kornia/kornia#489]
The function of ```adjust_hue``` can visually reproduce the similar results against Torchvision, although there may have two potential improvements to better align the Torchvision.
```python
        # TODO: HUE factor range:
        # In Kornia, we used (-pi, pi), while PIL used (-0.5, 0.5)
        # the range need to be re-normalized from (-0.5, 0.5) to (-pi, pi)
        # TODO: align implementation with pytorch, this might be the cause of inconsistency
        # Kornia: h_out: torch.Tensor = torch.fmod(h + hue_factor, divisor)
        # Torchvision: np_h += np.uint8(hue_factor * 255)
```

To reproduce:
```python
from PIL import Image
import requests
from io import BytesIO
from torch import allclose
from kornia.color.adjust import (
    adjust_brightness, adjust_contrast, adjust_saturation, adjust_hue)

response = requests.get("https://tinypng.com/images/social/website.jpg")
img = Image.open(BytesIO(response.content))

import torch

import kornia.augmentation.functional as F
from torchvision.transforms import functional as tvF
from torchvision.transforms import transforms

import matplotlib.pyplot as plt
import numpy as np

to_tensor = transforms.ToTensor()
to_pil = transforms.ToPILImage()

def tensor_pre_transform_wrapper(input: torch.Tensor):
    """ A wrapper that tried to reproduce the actual output from:
        - transforms.ToPILImage()
        - transforms.ToTensor()
        For each image, simply (img * 255).int() // 255
    """
    return (input * 255).int().float() / 255

in_tensor = to_tensor(img)
in_pil = img
factor = 0.5
out_tensor = adjust_hue(tensor_pre_transform_wrapper(in_tensor), torch.tensor(factor * 2 * 3.1415))
out_pil = tvF.adjust_hue(in_pil, factor)
```

<img width="402" alt="Screenshot 2020-03-21 at 10 46 33 AM" src="https://user-images.githubusercontent.com/15955486/77217935-47462300-6b61-11ea-8929-bed7ec34c7fe.png">

**Comment by edgarriba** [comment]
@shijianjian how far are the results ? for the `pi` factor , as far as I remember with @priba we decided to go with it since `pi` is a more representative number that just a factor as they do in torchvision.

**Comment by shijianjian** [comment]
It is not about PI factor I suppose, using a simple normalization like factor * 2 / PI works and produced very similar images. ```adjust_hue``` behaves very similar to ```to_grayscale``` in the distances between points. Still, I think it is about the precision format issue as documented in #490.

If we wish to reproduce PIL, we need to give a uint8 in every intermedia steps.

**Comment by edgarriba** [comment]
Then maybe we need some specific unit tests if we now the maths underlying under the function.

**Comment by shijianjian** [comment]
Yes. In order to do so, we have to align all functions to PIL implementations. Starting from:
- to_grayscale
- rgb_to_hsv
- hsv_to_rgb
then:
- adjust_hue
- adjust_brightness
- ...
We are basically re-writing all intensity-based conversions.
